ts/utils/navigation.ts

```diff
@@ -2,54 +2,46 @@
 
 import { index } from "fp-ts/lib/Array";
 import { fromNullable, none, Option } from "fp-ts/lib/Option";
-import { NavigationRoute } from "react-navigation";
+import {
+  NavigationLeafRoute,
+  NavigationRoute,
+  NavigationState
+} from "react-navigation";
 import { NavigationHistoryState } from "../store/reducers/navigationHistory";
 
-// TODO: Need to be fixed https://www.pivotaltracker.com/story/show/170819360
-export function getCurrentRouteName(navNode: any): string | undefined {
-  if (!navNode) {
-    return undefined;
-  }
-  if (
-    navNode.index === undefined &&
-    navNode.routeName &&
-    typeof navNode.routeName === "string"
-  ) {
-    // navNode is a NavigationLeafRoute
-    return navNode.routeName;
+/**
+ * Assuming that each and every NavigationRoute will eventually lead
+ * to a NavigationLeafRoute.
+ */
+const findLeafRoute = (
+  branchOrLeaf: NavigationRoute
+): NavigationLeafRoute | undefined => {
```

Comment:
It seems can't return `undefined`
```suggestion
): NavigationLeafRoute => {
```

Reply:
see https://github.com/pagopa/io-app/pull/3329#discussion_r700027615